plugins/rebalance/rest.go

```diff
@@ -0,0 +1,341 @@
+		 * few nodes and failed in few others */
+		logger.WithFields(log.Fields{
+			"error":   err.Error(),
+			"volname": volname,
+		}).Error("failed to start rebalance on volume")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	rebalinfo, err = GetRebalanceInfo(volname)
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"error":   err.Error(),
+			"volname": volname,
+		}).Error("failed to save rebalance info for volume")
+	}
+
+	logger.WithField("volname", rebalinfo.Volname).Info("rebalance started")
+
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, rebalinfo.RebalanceID)
+}
+
+func rebalanceStopHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	// collect inputs from url
+	volname := mux.Vars(r)["volname"]
+
+	lock, unlock := transaction.CreateLockFuncs(volname)
+	if err := lock(ctx); err != nil {
+		if err == transaction.ErrLockTimeout {
+			restutils.SendHTTPError(ctx, w, http.StatusConflict, err)
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		}
+		return
+	}
+	defer unlock(ctx)
+
+	// Validate rebalance command
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound)
+		return
+	}
+
+	rebalinfo, err := GetRebalanceInfo(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, ErrRebalanceNotStarted)
+		return
+	}
+
+	// Check rebalance process is started or not
+	if rebalinfo.Status != rebalanceapi.Started {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, ErrRebalanceNotStarted)
+		return
+	}
+
+	//A simple transaction to stop rebalance
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+
+	txn.Nodes = vol.Nodes()
+	txn.Steps = []*transaction.Step{
+		{
+			DoFunc: "rebalance-stop",
+			Nodes:  txn.Nodes,
+		},
+		{
+			DoFunc: "rebalance-store",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
+		},
+	}
+
+	err = txn.Ctx.Set("volname", volname)
+	if err != nil {
+		logger.WithError(err).Error("failed to set volname in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	rebalinfo.Volname = volname
+	rebalinfo.Status = rebalanceapi.Stopped
+	rebalinfo.Cmd = rebalanceapi.CmdStop
+
+	err = txn.Ctx.Set("rinfo", rebalinfo)
+	if err != nil {
+		logger.WithError(err).Error("failed to set rebalance info in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	err = txn.Do()
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"error":   err.Error(),
+			"volname": volname,
+		}).Error("failed to stop rebalance on volume")
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	txn.Ctx.Logger().WithField("volname", rebalinfo.Volname).Info("rebalance stopped")
```

Comment:
You can use "logger" instead of txn.Ctx.Logger() since this log message doesn't belong to transaction context